packages/playwright-core/src/utils/platform.ts

```diff
@@ -14,36 +14,56 @@
  * limitations under the License.
  */
 
+import * as crypto from 'crypto';
 import * as fs from 'fs';
 import * as path from 'path';
 import * as util from 'util';
 
 export type Platform = {
+  calculateSha1(text: string): Promise<string>;
+  createGuid: () => string;
   fs: () => typeof fs;
-  path: () => typeof path;
   inspectCustom: symbol | undefined;
+  path: () => typeof path;
   ws?: (url: string) => WebSocket;
 };
 
-export const emptyPlatform: Platform = {
-  fs: () => {
-    throw new Error('File system is not available');
-  },
-
-  path: () => {
-    throw new Error('Path module is not available');
+export const nodePlatform: Platform = {
+  calculateSha1: (text: string) => {
+    const sha1 = crypto.createHash('sha1');
+    sha1.update(text);
+    return Promise.resolve(sha1.digest('hex'));
   },
 
-  inspectCustom: undefined,
-};
+  createGuid: () => crypto.randomBytes(16).toString('hex'),
 
-export const nodePlatform: Platform = {
   fs: () => fs,
-  path: () => path,
+
   inspectCustom: util.inspect.custom,
+
+  path: () => path,
 };
 
 export const webPlatform: Platform = {
-  ...emptyPlatform,
+  calculateSha1: async (text: string) => {
+    const bytes = new TextEncoder().encode(text);
```

Comment:
Why can't we use the same implementation in Node.js?

Reply:
Because committee has been a bit too creative to my taste :)